### DISCUSSION TITLE
Toolbar Problem

### DISCUSSION THREAD

**Discussion by hoffst** [sumatrapdfreader/sumatrapdf#4373] [Q&A]
When I start SumatraPDF 3.5.2 with Parameter "-plugin" the first Toolbar-Button "Open File" is disabled. Please replace this Button with "Save as" like in former Versions e.g. 3.1.1 or make the Toolbar customizable. This seems to be a bug because "Save as" is accessible via CTRL+S.

**Comment by GitHubRulesOK** [comment]
the "plug-in" was by design for web use and restricted so in theory should not be able to access the file system the icon **was** as far as I recolect used for OPEN FILES in FOLDER  it should historically have got its file feed from the Netscape Browser, and still does in the few remaining up-to-date browsers it still runs in.

in previous versions like 3..3 that icon should perhaps be grey-ed out 
![image](https://github.com/user-attachments/assets/67e14eb2-5dd7-4dbc-b0cc-adfcb4b86532)



![image](https://github.com/user-attachments/assets/047c097b-3b23-4767-9758-715d383f660d)

2016 https://github.com/sumatrapdfreader/sumatrapdf/pull/511#issuecomment-216709805
>Plugin mode was created for embedding inside the browser as NPAPI plugin, so that you can open a PDF directly in the browser when clicking a link to a PDF file.

>The whole code should be removed because we no longer support NPAPI plugin mode because browsers no longer support NPAPI plugins.

**However** a few Browser still do and **Without Official Support** can either use Adobe Reader DC ActiveX (Like MSEdge in legacy Bing/Trident mode does) **OR** with Geko/Mozilla code use SumatraPDF NPAPI as a faster alternative

**Comment by hoffst** [comment]
Sad to hear that you intend to remove code for the command line parameter "-plugin". I just successfully managed to view PDF-files in a window of my own App. I consider this option as very useful, because it is still fully functional, very fast and needs only little resources. Do you offer or plan an "official" interface to embed SumatraPDF-Viewer in other Apps?

**Reply by GitHubRulesOK** [reply]
I am not the developer and also would like the functionality kept for those few browsers that maintain NPAPI compatability along with ActiveX but Adobe caused the problem with SWF that the NPAPI tech was sadly considered unsecure for Adobe PDF when a secure app like SumatraPDF was ignored !!

Technically SumatraPDF does not encourage embedding and many pirate forks exist that taint SumatraPDF success by not open sourcing their modifications nor feedback support..

**Comment by kjk** [comment]
Toolbar is now customizable. See https://www.sumatrapdfreader.org/docs/Customize-toolbar
